mito-ai/mito_ai/utils/telemetry_utils.py

```diff
@@ -126,3 +127,75 @@ def log(
 
     # TODO: Eventually we want to hook this up to the mito log uploader 
     # so enterprises can log usage if they want to.
+
+
+def log_ai_completion_success(
+    key_type: str, 
+    input_location: str, 
+    last_message_content: str, 
+    response: Dict[str, Any]
+) -> None:
+    """
+    Logs AI completion success based on the input location.
+
+    Args:
+        input_location: The location where the AI completion was triggered
+        last_message_content: The last message sent to the AI
+        response: The response received from the AI
+    """
+
+    code_cell_input = json.dumps(
+        last_message_content.split("Code in the active code cell:")[-1]
+        .strip()
+        .split("```python")[1]
+        .strip()
+        .split("```")[0]
+    )
+
+    if input_location == "smartDebug":
+        error_message = (
+            last_message_content.split("Error Message:")[-1]
+            .split("ERROR ANALYSIS:")[0]
+            .strip()
+        )
+        error_type = error_message.split(": ")[0]
+
+        log(
+            "smart_debug_success",
+            params={
+                KEY_TYPE_PARAM: key_type,
+                "code_cell_input": code_cell_input,
+                "error_message": error_message,
+                "error_type": error_type,
+                "response": response,
```

Comment:
Hmm so actually, it looks like Mixpanel has a [255 character limit for string ](https://docs.mixpanel.com/docs/data-structure/property-reference/data-type#string) and so I'm not actually seeing the code that we respond with in this response because in the smart debug, we respond with some thought tokens first. But even long code cells for the chat input would get cut off. 

That's a bummer. 

It looked into it, and Segment does not have the same limitation, so it's just Mixpanel that is bottlenecking us right now.

Reply:
Maybe we will eventually move away from Mixpanel for our Ai logging... but for now to keep us on track. Maybe we can use a function like this for the prompt and response. 

This function just takes a response and breaks it into chunks of 250 characters with the fields response_part_1, response_part_2, etc.

Thanks Claude :) 

```python
def chunk_response(response):
    """
    Split a response string into chunks of 250 characters.
    
    Args:
        response: The string to be chunked
        
    Returns:
        dict: A dictionary with keys 'response_part_1', 'response_part_2', etc.
    """
    chunk_size = 250
    chunks = {}
    
    # Handle empty or None response
    if not response:
        return {}
    
    # Calculate number of chunks needed
    num_chunks = (len(response) + chunk_size - 1) // chunk_size
    
    # Create chunks
    for i in range(num_chunks):
        start = i * chunk_size
        end = min(start + chunk_size, len(response))
        chunks[f'response_part_{i + 1}'] = response[start:end]
    
    return chunks
```

Reply:
And while we're doing this we can remove `{"completion":` from the logged response